packages/form/src/TextInputRow.js

```diff
@@ -95,55 +70,51 @@ class TextInputRow extends PureComponent {
         this.props.onBlur(event);
     }
 
-    handleInputChange = (event) => {
-        this.updateInputHeight(event.target);
-        this.props.onChange(event);
-    }
-
     renderInput({
         multiLine,
-        style: inputStyle = {},
-        ...inputProps
+        style = {},
+        renderInput,
+        inputComponent,
+        minRows,
+        maxRows,
+        ...otherProps
     }) {
         const sharedProps = {
-            ref: this.setInputRef,
             className: BEM.input.toString(),
+            style,
             onFocus: this.handleInputFocus,
             onBlur: this.handleInputBlur,
+            ...otherProps
         };
 
-        const textareaStyle = {
-            ...inputStyle,
-            height: this.state.inputHeight,
+        if (multiLine) {
+            const textareaProps = {
+                ...sharedProps,
+                minRows,
+                maxRows,
+            };
+
+            return <AutoSizeTextarea {...textareaProps} />;
+        }
+
+        const inputProps = {
+            type: 'text',
+            ...sharedProps,
         };
 
-        if (multiLine) {
-            return (
-                <textarea
-                    style={textareaStyle}
-                    onChange={this.handleInputChange}
-                    {...sharedProps}
-                    {...inputProps} />
-            );
+        if (isFunction(renderInput)) {
```

Comment:
What if it's used this way:

```jsx
<TextInputRow
    multiLine
    renderInput={func}
/>
```

Do you intent to let `renderInput` overridden by `multiLine`?

Reply:
No, input can't be customize when `multiLine` mode.